cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -923,62 +923,138 @@ PyMODINIT_FUNC PyInit_simulator(void)
+    },
     {"get_signal_val_real",
-        (PyCFunction)get_signal_val_real, METH_NOARGS,
-        "Get the value of a signal as a double precision float"},
+        (PyCFunction)get_signal_val_real, METH_NOARGS, PyDoc_STR(
+            "get_signal_val_real($self)\n"
+            "--\n\n"
+            "Get the value of a signal as a double precision float"
+        )
+    },
     {"set_signal_val_long",
-        (PyCFunction)set_signal_val_long, METH_VARARGS,
-        "Set the value of a signal using a long"},
+        (PyCFunction)set_signal_val_long, METH_VARARGS, PyDoc_STR(
+            "set_signal_val_long($self, action, value, /)\n"
+            "--\n\n"
+            "Set the value of a signal using a long"
+        )
+    },
     {"set_signal_val_str",
-        (PyCFunction)set_signal_val_str, METH_VARARGS,
-        "Set the value of a signal using an NUL-terminated 8-bit string"},
+        (PyCFunction)set_signal_val_str, METH_VARARGS, PyDoc_STR(
+            "set_signal_val_str($self, action, value, /)\n"
+            "--\n\n"
+            "Set the value of a signal using an NUL-terminated 8-bit string"
+        )
+    },
     {"set_signal_val_binstr",
-        (PyCFunction)set_signal_val_binstr, METH_VARARGS,
-        "Set the value of a signal using a string with a character per bit"},
+        (PyCFunction)set_signal_val_binstr, METH_VARARGS, PyDoc_STR(
+            "set_signal_val_binstr($self, action, value, /)\n"
+            "--\n\n"
+            "Set the value of a signal using a string with a character per bit"
+        )
+    },
     {"set_signal_val_real",
-        (PyCFunction)set_signal_val_real, METH_VARARGS,
-        "Set the value of a signal using a double precision float"},
+        (PyCFunction)set_signal_val_real, METH_VARARGS, PyDoc_STR(
+            "set_signal_val_real($self, action, value, /)\n"
+            "--\n\n"
+            "Set the value of a signal using a double precision float"
+        )
+    },
     {"get_definition_name",
-        (PyCFunction)get_definition_name, METH_NOARGS,
-        "Get the name of a GPI object's definition"},
+        (PyCFunction)get_definition_name, METH_NOARGS, PyDoc_STR(
+            "get_definition_name($self)\n"
+            "--\n\n"
+            "Get the name of a GPI object's definition"
+        )
+    },
     {"get_definition_file",
-        (PyCFunction)get_definition_file, METH_NOARGS,
-        "Get the file that sources the object's definition"},
+        (PyCFunction)get_definition_file, METH_NOARGS, PyDoc_STR(
+            "get_definition_file($self)\n"
+            "--\n\n"
+            "Get the file that sources the object's definition"
+        )
+    },
     {"get_handle_by_name",
-        (PyCFunction)get_handle_by_name, METH_VARARGS,
-        "Get handle of a named object"},
+        (PyCFunction)get_handle_by_name, METH_VARARGS, PyDoc_STR(
+            "get_handle_by_name($self, name, /)\n"
+            "--\n\n"
+            "Get handle of a named object"
```

Comment:
```suggestion
            "Get handle to a child object by name"
```
This now works by getting a handle to a sub-object only, right?